### PR TITLE
Testing historical component

### PR DESCRIPTION
**Pull request by J-Poulter**
# PR Description

Please include a summary of the change:

On this branch I wrote the tests for the historical and results components.  The test coverage so far is 70% for the Historical and 100% for the Results.  If we have the time, I plan on coming back to the Historical component to cover more of the untested parts.  Also, in the Historical component JS I was able to find a way to dynamically use one hook to update all of the month inputs.  Unsure yet if this will be implemented or if we will be sticking with the previous approach of using a hook for each month input.

## Type of change
Please uncheck options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Add Testing
- [X] Refactor

## Checklist:
- [X] My code follows the Turing style guidelines
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/containers/Historical/Historical.js [hunk 1]
```diff
@@ -6,20 +6,26 @@ import { Redirect } from 'react-router-dom'
...
+  // const [march, updateMarch] = useState(0)
+  // const [april, updateApril] = useState(0)
+  // const [may, updateMay] = useState(0)
+  // const [june, updateJune] = useState(0)
+  // const [july, updateJuly] = useState(0)
+  // const [august, updateAugust] = useState(0)
+  // const [september, updateSeptember] = useState(0)
+  // const [october, updateOctober] = useState(0)
+  // const [november, updateNovember] = useState(0)
+  // const [december, updateDecember] = useState(0)
```
Comment (alex-latham) [inline src/containers/Historical/Historical.js:20]: Probably best to have these set to zero for initial state.

File: src/containers/Historical/Historical.js [hunk 2]
```diff
@@ -63,8 +69,8 @@ const Historical = ({ userProfile, userQuote, loadSolarData, history }) => {
         'Content-Type': 'application/json'
       },
       body: JSON.stringify({
-        solarizer_parameters: solarizer_parameters,
-        // historical_kWh: historical_kWh
+        solarizer_parameters,
+        // historical_kWh
```
Comment (alex-latham) [inline src/containers/Historical/Historical.js:73]: These shortcuts are neat.

File: src/containers/Historical/Historical.js (hunk 1 above)
Comment (nickstaylor) [reply src/containers/Historical/Historical.js:20]: Let's look at this next

File: src/containers/Historical/Historical.js [hunk 3]
```diff
@@ -6,20 +6,26 @@ import { Redirect } from 'react-router-dom'
...
+  // const [october, updateOctober] = useState(0)
+  // const [november, updateNovember] = useState(0)
+  // const [december, updateDecember] = useState(0)
   const [historicalData, updateHistoricalData] = useState(false)
 
+  const [dataEntered, setDataEntered] = useState({january: ''})
+
+  const changeValue = (event) => {
+    setDataEntered({ ...dataEntered, [event.target.name]: parseFloat(event.target.value) })
+  }
```
Comment (nickstaylor) [inline src/containers/Historical/Historical.js:27]: great refactoring here!

**Pull request merged**